avaje-jex/src/main/java/io/avaje/jex/jdk/RoutingHandler.java

```diff
@@ -30,26 +33,14 @@ void waitForIdle(long maxSeconds) {
   }
 
   @Override
-  public void doFilter(HttpExchange exchange, Filter.Chain chain) {
+  public void handle(HttpExchange exchange) {
     final String uri = exchange.getRequestURI().getPath();
     final Routing.Type routeType = mgr.lookupRoutingType(exchange.getRequestMethod());
     final SpiRoutes.Entry route = routes.match(routeType, uri);
 
     if (route == null) {
       var ctx = new JdkContext(mgr, compressionConfig, exchange, uri, Set.of());
-      routes.inc();
-      try {
-        ctx.setMode(Mode.BEFORE);
-        processNoRoute(ctx, uri, routeType);
-        exchange.setAttribute("JdkContext", ctx);
-        chain.doFilter(exchange);
-        handleNoResponse(exchange);
-      } catch (Exception e) {
-        handleException(ctx, e);
-      } finally {
-        routes.dec();
-        exchange.close();
-      }
+      handleException(ctx, new NotFoundException("uri: " + uri));
```

Comment:
Looks like this removes support for HEAD requests where there is a GET route

Reply:
It seems the point of a HEAD request is to get [metadata for a GET request via headers](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/HEAD). Just having a blank 200 without any additional info doesn't seem to really cut it.